### PR TITLE
Merge FeAI into DeAI

### PR DESCRIPTION
**Pull request by s314cy**
* Merge the two servers.
* Keep all the UI from the already-done DeAI-FeAI merge.
* Put FeAI-specific backend into temporary subfolders.
* Adapt DeAI-specific backend to the changes in the UI and backend.
* Keep the remaining backend files from FeAI.

### CODE REVIEW

**Review by Nacho114 — APPROVED** [review]
- Using an abstract client class the app can choose between FeAI and DeAI communication schemes which allows the integration of the apps.
- The merge was done with speed as a priority in order to reduce divergence between the apps; this means there is still work to be done to make the integration cleaner.

Has been tested by Hugo, Laurynas and Ignacio and it seems to be stable for more people to test/use.

File: mobile-browser-based-version/server/README.md [hunk 1]
```diff
@@ -1,22 +1,61 @@
...
+  - client receives the number of data samples per client ID for communication round `round` and task `task`
+    ```
+    URL: /receive_data_info/<task>/<round>
+    Body: { id: String, timestamp: Date }
+    ```
 
 #### Tasks
 
-The training tasks given to DeAI clients are centralized on this server. Their descriptions as well as their deep learning model architectures must be made available to all peers, which is achieved via the following routing paths:
+The training tasks given to FeAI clients are centralized on this server. Their descriptions as well as their deep learning model architectures must be made available to all clients, which is achieved via the following `GET` request routes.
```
[the remark this answers is not in the document]
Comment (s314cy) [reply mobile-browser-based-version/server/README.md:49]: i'm merging the two readmes within a single file :) so the final readme contains information on both the feai server and the deai server

[the remark this answers is not in the document]
Comment (Nacho114) [reply mobile-browser-based-version/server/README.md:49]: @s314cy is working on merging the Readme of FeAI and DeAi

File: mobile-browser-based-version/server/src/router/router.js [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+        target: `${config.SERVER_URI}:${port}`,
+        changeOrigin: true,
+        ws: true,
+      })
+    );
+  })
+);
+
+decentralisedRouter.use('/tasks', tasksRouter);
+decentralisedRouter.get('/', (req, res) => res.send('DeAI server'));
```
[the remark this answers is not in the document]
Comment (Nacho114) [reply mobile-browser-based-version/server/src/router/router.js:85]: Yeah this is a good idea, will add to the todo list (which will include important issues).

**Review by martinjaggi — APPROVED** [review]
ok let's merge this tiny PR ;).

great work, thanks all!

**Pull request merged**